core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadOnlyLDAPUserStoreManager.java

```diff
@@ -1656,6 +1659,179 @@ public Group doGetGroupFromGroupName(String groupName, List<String> requiredAttr
         return buildGroupFromAttributes(responseAttributes, attributesByLDAPSearch);
     }
 
+    @Override
+    public List<Group> doListGroups(Condition condition, int limit, int offset, String sortBy, String sortOrder)
+            throws UserStoreException {
+
+        List<ExpressionCondition> expressionConditions = getExpressionConditions(condition);
+        if (expressionConditions.isEmpty()) {
+            throw new UserStoreClientException(ERROR_EMPTY_GROUP_SEARCH_FILTER.getMessage(),
```

Comment:
Shouldn't we call listGroup() without filter, rather throwing an exception?

Reply:
Fixed